mitodl/mit-open
# Server render search results (search and channel pages)

### What are the relevant tickets?





Closes: 
[Next.js: Search page: Prefetch API calls #6035](https://github.com/mitodl/hq/issues/6035)
[Next.js: Channel details page search: Prefetch API calls #6037](https://github.com/mitodl/hq/issues/6037)

### Description (What does it do?)


Prefetches search results for the search and channel details pages.

Dependency on https://github.com/mitodl/mit-learn/pull/1822 to merge first and change base to main.

This also needs https://github.com/mitodl/course-search-utils/pull/159 to be published before merging.


### How can this be tested?


On the search page /search and channel pages e.g. /c/topic/architecture, /c/department/media-arts-and-sciences, /c/unit/mitx, make various changes to the following, refreshing after each and ensuring that the search results are prerendered on the server with no prefetch warnings in the dev tools console:

- Search q text
- Filter facet selections
- Resource category tab selection
- Page selection

The search results should render ok on first load with JavaScript disabled in the dev tools debugger settings.

### Additional Context



Includes some refactoring to pull out `getSearchParams()` for reuse and provide a `requestSearchParams()` to convert Next.js parsed querystrings to our params for `learningResources.search()` API calls.
